bundles/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeDimmerHandler.java

```diff
@@ -0,0 +1,227 @@
+                    updateDimmerOutputIntensity(outputIntensity);
+                    logger.debug("Output intensity is now: {} %%", answOutputIntensity);
+
+                } else {
+                    logger.debug("Cannot set output intensity, status: {}", answ.getStatus());
+                }
+            } else {
+                logger.debug("Could not connect to bridge to update Output Intensity");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot connect to bridge");
+            }
+        } finally {
+            this.getSinopeGatewayHandler().schedulePoll();
+        }
+    }
+
+    private void setLightMode(int mode) throws UnknownHostException, IOException {
+        getSinopeGatewayHandler().stopPoll();
+        try {
+            if (getSinopeGatewayHandler().connectToBridge()) {
+                logger.debug("Connected to bridge");
+
+                SinopeDataWriteRequest req = new SinopeDataWriteRequest(getSinopeGatewayHandler().newSeq(), deviceId,
+                        new SinopeLightModeData());
+                ((SinopeLightModeData) req.getAppData()).setLightMode((byte) mode);
+
+                SinopeDataAnswer answ = (SinopeDataAnswer) getSinopeGatewayHandler().execute(req);
+
+                if (answ.getStatus() == DATA_ANSWER) {
+                    int answLightMode = (((SinopeLightModeData) answ.getAppData())).getLightMode();
+                    updateLightMode(answLightMode);
+                    logger.debug("Light mode is now : {}", answLightMode);
+                } else {
+                    logger.debug("Cannot Set Light mode, status: {}", answ.getStatus());
+                }
+            } else {
+                logger.debug("Could not connect to bridge to update Light Mode");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Cannot connect to bridge");
+            }
+        } finally {
+            getSinopeGatewayHandler().schedulePoll();
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged {}", bridgeStatusInfo);
+        updateDeviceId();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initializeThing thing {}", getThing().getUID());
+        updateDeviceId();
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        updateDeviceId();
+    }
+
+    public void updateDimmerOutputIntensity(int outputIntensity) {
+        updateState(SinopeBindingConstants.CHANNEL_DIMMER_OUTPUTINTENSITY, new PercentType(outputIntensity));
+    }
+
+    public void updateLightMode(int lightMode) {
+        updateState(SinopeBindingConstants.CHANNEL_LIGHTMODE, new DecimalType(lightMode));
+    }
+
+    public void update() throws IOException {
+        if (this.deviceId != null) {
```

Comment:
Is this check needed? After `initialize()` deviceId should be set as this is a required parameter. When calling this method from the bridge handler it might be needed to check if thing is initialized though,